Userland/Libraries/LibGfx/ImageFormats/ILBMLoader.cpp

```diff
@@ -211,6 +236,22 @@ static ErrorOr<void> extend_ehb_palette(ILBMLoadingContext& context)
     return {};
 }
 
+static ErrorOr<void> reduce_ham_palette(ILBMLoadingContext& context)
+{
+    u8 bits = context.cmap_bits;
+
+    dbgln_if(ILBM_DEBUG, "reduce palette planes={} bits={}", context.bm_header.planes, context.cmap_bits);
+
+    if (bits > context.bm_header.planes) {
+        dbgln_if(ILBM_DEBUG, "need to reduce palette");
+        bits -= (bits - context.bm_header.planes) + 2;
+        context.color_table.resize((context.color_table.size() >> bits));
```

Comment:
Would it be worth adding some checks that `bm_header.planes` and `context.color_table.size()`  are the values we expect if `has_flag(context.viewport_mode, ViewportMode::HAM)` is true? This would ensure that these calculations can't underflow.

Reply:
I'm not sure what you mean?

Reply:
Sorry, I didn't explain this very well.

It looks like this calculation could underflow if `bits < (bits - context.bm_header.planes) + 2`.

Looking at it again though, I'm not sure how much of a problem this is, as the values would have to be pretty nonsensical to hit this condition. Decoding would likely fail shortly after this point anyway. This would purely be to guard against malformed files.